tests/core/test_circuit.py

```diff
@@ -350,3 +353,42 @@ def test_to_qasm_3(circuit: QCircuit, printed_result_filename: str):
         encoding="utf-8",
     ) as f:
         assert circuit.to_qasm3().strip() == f.read().strip()
+
+
+@pytest.mark.parametrize(
+    "circuit, expected_matrix",
+    [
+        (QCircuit([H(0)]), H(0).to_matrix()),
+        (QCircuit([CNOT(0, 1)]), CNOT(0, 1).to_matrix()),
+        (
+            QCircuit([H(0), CNOT(0, 1)]),
+            np.dot(
+                CNOT(0, 1).to_matrix(), np.kron(H(0).to_matrix(), Id(0).to_matrix())
+            ),
+        ),
+        (QCircuit([Id(0)]), Id(0).to_matrix()),
+        (QCircuit([SWAP(0, 1)]), SWAP(0, 1).to_matrix()),
+        (QCircuit([]), np.array([[1]])),
+        (
+            QCircuit([Rz(np.pi / 4, 0)]),
+            Rz(np.pi / 4, 0).to_matrix(),
+        ),
+        (
+            QCircuit([Ry(np.pi / 2, 0)]),
+            Ry(np.pi / 2, 0).to_matrix(),
+        ),
+    ],
+)
+def test_to_matrix(circuit: QCircuit, expected_matrix: Matrix):
```

Comment:
Should we add more complicated circuit with many more gates, for which we computed the matrix using Maple or Mathematica for instance ?

Reply:
for now the qcircuit .to_matrix is still the qiskit version

Reply:
Ok makes sense then